web/src/app/core/services/activity.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { Subject, Observable, of, timer } from 'rxjs';
+import { finalize, startWith } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ActivityService {
+
+  private counter: number = 0;
+  private subject: Subject<number> = new Subject();
+
+  // This function will set the progress bar status
+  public setProgressBar(status: boolean): number {
+    switch(status) {
+        case true:
+          this.counter++;
+        break;
+        case false:
+          this.counter--;
+        break;
+    }
+
+    // activity bar disappears too quickly, delay the final step
+    if (this.counter === 0 && status === false) {
+        timer(500).subscribe(() => this.subject.next(this.counter));
+    } else {
+        this.subject.next(this.counter);
+    }
+
+    return this.counter;
+  }
+
+  // this function will return the status of progress bar to main component
+  public getProgressBar(): Observable<number> {
+    return this.subject.asObservable();
+  }
+
+  public start(): Observable<number> {
```

Comment:
Please add comment on function declaration. I think we should start using compdocs for function declarations .
Later on we can generate the documentation of our project using compdoc
https://compodoc.app/

We can start new issue for project documentation . What do you think ?

Reply:
This looks like a good idea 👍 . Yes we will need a new issue to go through and update existing code, then going forward we can do it as we go along 🤓

Reply:
Created #1248

Reply:
Great 👍